extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -123,13 +148,32 @@ function SidebarWithRef(
     [open, close, toggle]
   );
 
+  useLayoutEffect(() => {
+    if (side) {
+      return;
+    }
+    const sidebarElement = sidebarRef.current;
+    if (!sidebarElement) {
+      return;
+    }
+    setSide(calculateSide(sideProp, sidebarElement.ownerDocument));
+  }, [sideProp, side, mounted]);
+
   useLayoutEffect(() => {
     const sidebarElement = sidebarRef.current;
     const backdropElement = backdropRef.current;
     if (!sidebarElement || !backdropElement) {
       return;
     }
 
+    if (!side) {
+      safelySetStyles(sidebarElement, SIDE_STYLES_HIDDEN);
+      safelySetStyles(backdropElement, SIDE_STYLES_HIDDEN);
+      return;
+    }
+    safelySetStyles(sidebarElement, SIDE_STYLES_VISIBLE);
+    safelySetStyles(backdropElement, SIDE_STYLES_VISIBLE);
```

Comment:
One the first render, `side` will not yet be calculated.  BUT the sidebar and mask elements are already rendered.  We need to hide them to prevent FOUC.  Then return.

One the second render `side` has been calculated in the previous render.  We can make the sidebar and mask elements visible and begin animations.

Reply:
I tend to prefer setting styles declaratively over imperatively when possible. What do you think about setting styles directly in a hook rather than doing this in the render markup, i.e. `{mounted && side && <>...</>}`? Another declarative approach could be to apply `hidden={!side}` on both elements in the JSX. 

I see the sidebar already does much of the imperative logic for animations, but I had assumed that was intended to be minimal because there was no other way around it.

Reply:
Oh `hidden={!side}` might work.  `{mounted && side && <>...</>}` unfortunately won't work since we still want to render the elements when side is `falsy` in order to get the elementRef and then do the calculation on side.

I will think about this and follow up in the next revision.

Reply:
Used `hidden={!side}` in the latest.